api/api.py

```diff
@@ -66,18 +66,17 @@ async def store_event(
         producer_key = key_manager.generate_key()
         logger.debug(f"Producer key: {producer_key}")
 
-        # (4) Parse general properties config to later get kafka_topic from it
-        general_config_dict = config["general"]
+        kafka_topic = config.app.kafka_topic
 
         # (5) Send serialized_event to Kafka
         kafka_producer.produce(
-            topic=general_config_dict["kafka_topic"],
+            topic=kafka_topic,
             value=serialized_event,
             key=producer_key,
             on_delivery=delivery_report,
         )
-        # Ensure messages are sent before returning
-        kafka_producer.flush(timeout=20)  
+        # Trigger the on_delivery callback
+        kafka_producer.poll(0) 
```

Comment:
in order to see the effect of `on_delivery` callback, instead flushing accumulated batch, we can just run [poll](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.poll), which doesn't trigger sending messages to broker, so we don't interfere with kafka client producer async logic and don't hit performance

Reply:
I supposet this should be a [poll](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Producer.poll) method for producer (not consumer)